lightning/src/ln/peer_handler.rs

```diff
@@ -757,7 +758,19 @@ impl<Descriptor: SocketDescriptor, CM: Deref, RM: Deref, L: Deref, CMH: Deref> P
 	fn enqueue_message<M: wire::Type>(&self, peer: &mut Peer, message: &M) {
 		let mut buffer = VecWriter(Vec::with_capacity(2048));
 		wire::write(message, &mut buffer).unwrap(); // crash if the write failed
-		log_trace!(self.logger, "Enqueueing message {:?} to {}", message, log_pubkey!(peer.their_node_id.unwrap()));
+
+		match message.type_id() {
+			msgs::Ping::TYPE |
+			msgs::Pong::TYPE |
+			msgs::ChannelAnnouncement::TYPE |
```

Comment:
This appears to be missing ChannelUpdate.

Reply:
Give BOLT 1 defines "routing" messages are in a specific message type range, maybe we should add a function/method to `wire` to prevent missing any messages and including any new ones.

https://github.com/lightning/bolts/blob/master/01-messaging.md

Reply:
You're right, a function along the lines of `fn is_gossip_msg(type_id: u16)` would probably simplify this. However, this assumes that we strictly want to log all gossip messages via the `Gossip` level, and no other messages, even if they could spam the log output. Is this the case, then I'll gladly do it this way?

Reply:
Yeah, I think so. Maybe a function named `is_gossip` on `wire::Type` with a default implementation would be sufficient.

Reply:
I dunno if we want to be super strict about it, but we could still unify the code - we probably actually want to log some of the filter/query stuff as TRACE since its not usually very noisy on "normal" connections. That said, we've never cared about trying to match LDK's interfaces up with the spec exactly, and dunno why we should start here. Maybe `is_gossip_msg` isn't the right name, maybe something like `is_gossip_update_msg` or something else to communicate its one of channel/node announcement/update messages and not something asking for such messages.

Reply:
As suggested, https://github.com/rust-bitcoin/rust-lightning/pull/1145/commits/e3b4d90ad981331270b3fb9dd1c7743e8f60efc2 implements `is_gossip_update_msg` on `wire::Type`, which simplifies and future-proves the code in `PeerHandler`.

Reply:
BOLT 1 calls the entire range as "routing" messages. Then in BOLT 7 the announcement and update messages are the ones considered "gossip" messages while the remaining are consider gossip queries as defined by the `gossip_queries` feature bit. @TheBlueMatt Are you saying we should log the former as level `Gossip` and the latter as `Trace` like all the other messages?

Reply:
Yes, I think so. I don't really think there's any reason for this API to be tied to spec details. What we really want to do is make full trace logs not be GiB in size without removing any information that could potentially be useful. Queries could be important for debugging and also aren't very voluminous, so I suggested making the distinction "graph sync" messages, not just gossip ones.